pkg/contexthelpers/context_test.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2020 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build !privileged_tests
+
+package contexthelpers
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"gopkg.in/check.v1"
+)
+
+func Test(t *testing.T) {
+	check.TestingT(t)
+}
+
+type ContextSuite struct{}
+
+var _ = check.Suite(&ContextSuite{})
+
+func (b *ContextSuite) TestConditionalTimeoutContext(c *check.C) {
+	ctx, cancel, ch := NewConditionalTimeoutContext(context.Background(), 10*time.Millisecond)
+	c.Assert(ctx, check.Not(check.IsNil))
+	c.Assert(cancel, check.Not(check.IsNil))
+	c.Assert(ch, check.Not(check.IsNil))
+
+	select {
+	case <-ctx.Done():
+	case <-time.After(time.Second):
+		c.Errorf("no timeout despite reporting success")
```

Comment:
nit: Isn't this more like "conditional timeout was not triggered"?

I found this confusing as 'no timeout' could apply to this timeout from `time.After()` or the timeout configured in the NewConditionalTimeoutContext() call above, and also I don't understadn "despite reporting success" since this path through the test case doesn't send a success to the channel.